Microsoft.NET.Build.Containers/Layer.cs

```diff
@@ -95,6 +100,28 @@ public static Layer FromFiles(IEnumerable<(string path, string containerPath)> f
 
     private readonly static char[] PathSeparators = new char[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
 
+    /// <summary>
+    /// Ensures that all directory entries for the given segments are created within the tar.
+    /// </summary>
+    /// <param name="tar">The tar into which to add the directory entries.</param>
+    /// <param name="directoryEntries">The lookup of all known directory entries. </param>
+    /// <param name="filePathSegments">The segments of the file within the tar for which to create the folders</param>
+    private static void EnsureDirectoryEntries(TarWriter tar, ISet<string> directoryEntries, IReadOnlyList<string> filePathSegments)
```

Comment:
nit: can be better as local function, then passing `directoryEntries` is not needed. 
With current implementation `directoryEntries` is being modified, which is quite implicit.